octodns/provider/constellix.py

```diff
@@ -5,1113 +5,19 @@
-                                zone.name,
-                                record['type'],
-                                record['id'])
-                            # delete geofilter
-                            self.log.debug(
-                                "Deleting geofilter %s",
-                                zone.name)
-                            self._client.geofilter_delete(
-                                record['geolocation']['geoipFilter'])
-
-                            # delete pool
-                            self.log.debug(
-                                "Deleting pool %s %s",
-                                zone.name,
-                                record['type'])
-                            self._client.pool_delete(
-                                record['type'],
-                                record['pools'][0])
-
-                # for all the rest records
-                else:
-                    self._client.record_delete(
-                        zone.name, record['type'], record['id'])
-        # delete World Default
-        if world_default_record:
-            # delete record
-            self.log.debug(
-                "Deleting record %s %s",
-                zone.name,
-                world_default_record['type']
-            )
-            self._client.record_delete(
-                zone.name,
-                world_default_record['type'],
-                world_default_record['id']
-            )
-            # delete pool
-            self.log.debug(
-                "Deleting pool %s %s",
-                zone.name,
-                world_default_record['type']
-            )
-            self._client.pool_delete(
-                world_default_record['type'],
-                world_default_record['pools'][0]
-            )
-
-    def _apply(self, plan):
-        desired = plan.desired
-        changes = plan.changes
-        self.log.debug('_apply: zone=%s, len(changes)=%d', desired.name,
-                       len(changes))
-
-        try:
-            self._client.domain(desired.name)
-        except ConstellixClientNotFound:
-            self.log.debug('_apply:   no matching zone, creating domain')
-            self._client.domain_create(desired.name[:-1])
-
-        for change in changes:
-            class_name = change.__class__.__name__
-            getattr(self, f'_apply_{class_name}')(
-                change,
-                desired.name)
-
-        # Clear out the cache if any
-        self._zone_records.pop(desired.name, None)
+from logging import getLogger
+
+logger = getLogger('Constellix')
+try:
+    logger.warn('octodns_constellix shimmed. Update your provider class to '
+                'octodns_constellix.ConstellixProvider. '
+                'Shim will be removed in 1.0')
+    from octodns_constellix import ConstellixProvider, ConstellixBaseProvider
+    ConstellixProvider  # pragma: no cover
+    ConstellixBaseProvider  # pragma: no cover
```

Comment:
@ross This class does not exist (yet?) in https://github.com/octodns/octodns-constellix/blob/main/octodns_constellix/__init__.py
I would tend to drop the reference here.

Reply:
Doh. Copy-n-paste error. Will get a PR going shortly.

Reply:
https://github.com/octodns/octodns/pull/843